### PR TITLE
Add gke-operator

### PR DESCRIPTION
**Pull request by thedadams**
Related issue:
https://github.com/rancher/rancher/issues/32293

### CODE REVIEW

**Review by cmurphy — CHANGES_REQUESTED** [review]
I think this needs a `rancher-gke-operator-crd` package too?

**Comment by thedadams** [comment]
@cmurphy Yes, thanks for catching that.

**Review by cmurphy — DISMISSED** [review]

**Comment by cbron** [comment]
You'll need to rebase this onto new charts system, which should resolve the whole rc problem

**Review by aiyengar2 — COMMENTED** [review]
This PR needs to be moved to the dev-v2.6 branch as the dev-v2.6-source branch is now deprecated.

The easiest way to do this would be to create a new local branch from the `dev-v2.6` branch, cherry-pick your commits from the branch this PR points to to that new branch, and then run `make charts` to generate the charts/ and assets/ as part of the new provisioning workflow.

More details on the new workflow can be found in https://github.com/rancher/charts/blob/dev-v2.5/packages/README.md#common-workflow. Feel free to find me on Slack / RocketChat or email me at arvind.iyengar@suse.com for any questions.

**Review by cmurphy — COMMENTED** [review]
I think this needs a `generated-changes/` directory?

**Comment by thedadams** [comment]
@cmurphy Dang it, you're right. I was told that bugged was fixed now, but it doesn't seem to be.

File: charts/rancher-gke-operator-crd/rancher-gke-operator-crd/2.0.100-rc1+up2.0.1-rc1/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/hidden: "true"
+  catalog.cattle.io/namespace: cattle-system
+  catalog.cattle.io/os: linux
+  catalog.cattle.io/release-name: rancher-gke-operator-crd
+apiVersion: v2
+appVersion: 2.0.1-rc1
```
Comment (cmurphy) [inline charts/rancher-gke-operator-crd/rancher-gke-operator-crd/2.0.100-rc1+up2.0.1-rc1/Chart.yaml:8]: Is this right? Before this we would have had a generated-changes patch to make this `2.0.1`.

**Review by cmurphy — APPROVED** [review]

File: packages/rancher-gke-operator-crd/package.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+url: https://github.com/rancher/gke-operator/releases/download/v2.0.1-rc1/rancher-gke-operator-crd-2.0.1-rc1.tgz
```
Comment (cmurphy) [inline packages/rancher-gke-operator-crd/package.yaml:1]:
```suggestion
url: https://github.com/rancher/gke-operator/releases/download/v1.1.1-rc1/rancher-gke-operator-crd-1.1.1-rc1.tgz
```

File: packages/rancher-gke-operator/package.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+url: https://github.com/rancher/gke-operator/releases/download/v2.0.1-rc1/rancher-gke-operator-2.0.1-rc1.tgz
```
Comment (cmurphy) [inline packages/rancher-gke-operator/package.yaml:1]:
```suggestion
url: https://github.com/rancher/gke-operator/releases/download/v1.1.1-rc1/rancher-gke-operator-1.1.1-rc1.tgz
```

**Review by cmurphy — CHANGES_REQUESTED** [review]

**Review by cmurphy — APPROVED** [review]

**Review by cmurphy — DISMISSED** [review]

**Review by aiyengar2 — APPROVED** [review]
LGTM

File: index.yaml [hunk 4]
```diff
@@ -1309,6 +1309,119 @@ entries:
...
+    created: "2021-05-18T16:21:52.95982-07:00"
+    description: A Helm chart for provisioning GKE clusters
+    digest: 9ccb55efd79d1c5f608d349837ca978ddb590f1e639528dd62e7ce12cf2e8955
+    home: https://github.com/rancher/gke-operator
+    name: rancher-gke-operator
+    sources:
+    - https://github.com/rancher/gke-operator
+    urls:
+    - assets/rancher-gke-operator/rancher-gke-operator-2.0.100-rc1+up2.0.1-rc1.tgz
+    version: 2.0.100-rc1+up2.0.1-rc1
```
Comment (cbron) [inline index.yaml:1334]: This needs to be regenerated, shouldn't be adding gke 2.0 stuff in here

File: index.yaml [hunk 5]
```diff
@@ -1309,6 +1309,119 @@ entries:
...
+      catalog.cattle.io/release-name: rancher-gke-operator-crd
+    apiVersion: v2
+    appVersion: 1.1.1-rc1
+    created: "2021-05-21T13:36:21.572363-07:00"
+    description: GKE Operator CustomResourceDefinitions
+    digest: 6a4ea247a0d646da61e411874dcef9693b964bf3147069ee0dd51916dac152ea
+    name: rancher-gke-operator-crd
+    urls:
+    - assets/rancher-gke-operator-crd/rancher-gke-operator-crd-1.1.101-rc1+up1.1.1-rc1.tgz
+    version: 1.1.101-rc1+up1.1.1-rc1
```
Comment (cbron) [inline index.yaml:1409]: Don't get why we are listing 1.1.101 here either

**Review by cbron — CHANGES_REQUESTED** [review]
Chart looks good, but some index.yaml problems

File: index.yaml (hunk 4 above)
Comment (thedadams) [reply index.yaml]: Removed.

File: index.yaml (hunk 5 above)
Comment (thedadams) [reply index.yaml]: Removed.

**Review by cbron — APPROVED** [review]

**Pull request merged**